types.d.ts

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import * as Quill from "quill";
+
+type ToolbarOptionItem = string | string[] | Object[];
+
+interface ToolbarOptionObject {
+    container: string;
+    handlers?: {
+        [key: string]: Function;
+    };
+}
+
+export interface QuillModules {
+    toolbar: string | ToolbarOptionObject | ToolbarOptionItem[];
+}
+
+export interface UnprivilegedEditor {
+    getLength: () => number;
+    getText: () => string;
+    getHTML: () => string;
+    getBounds(index: number, length?: number): Quill.BoundsStatic;
+    getSelection(focus?: boolean): Quill.RangeStatic;
+}
+
+export interface QuillComponentProps {
+    id: string;
+    className: string;
+    theme: string;
+    style: React.CSSProperties;
+    readOnly: boolean;
+    value: string | Quill.Delta;
+    defaultValue: string | Quill.Delta;
+    placeholder: string;
+    tabIndex: number;
+    bounds: string | HTMLElement;
+    onChange(content: string, delta: Quill.Delta, source: Quill.Sources, editor: UnprivilegedEditor): void;
+    onChangeSelection(range: Quill.RangeStatic, source: Quill.Sources, editor: UnprivilegedEditor): void;
+    onFocus(range: Quill.RangeStatic, source: Quill.Sources, editor: UnprivilegedEditor): void;
+    onBlur(previousRange: Quill.RangeStatic, source: Quill.Sources, editor: UnprivilegedEditor): void;
+    onKeyPress: React.EventHandler<any>;
+    onKeyDown: React.EventHandler<any>;
+    onKeyUp: React.EventHandler<any>;
+    modules: QuillModules;
+    // toolbar: never;
+    formats: string[];
+    // styles: never;
+    // pollInterval: never;
+    children: React.ReactElement<any>;
+}
+
+export default class QuillComponent extends React.Component<QuillComponentProps> {
+    focus(): void;
+    blur(): void;
+    getEditor(): Quill.Quill;
+}
+
+export interface QuillToolbarItems {}
```

Comment:
~~~ts
export interface QuillToolbarItems {
  type: string,
  label?: string,
  value?: string.
  items?: QuillToolbarItems[],
};
~~~

Reply:
Also, this should be singular, ie. `QuillToolbarItem`.